lib/src/getContents/bufferFile.js

```diff
@@ -3,12 +3,18 @@
 var fs = require('graceful-fs');
 var stripBom = require('strip-bom');
 
-function bufferFile(file, cb) {
-  fs.readFile(file.path, function(err, data) {
+function bufferFile(file, opt, cb) {
+  fs.readFile(file.path, opt, function(err, data) {
     if (err) {
       return cb(err);
     }
-    file.contents = stripBom(data);
+    // if user specifies encoding type, a string is returned
+    // so we need to convert it back to buffer
+    if (!Buffer.isBuffer(data) && typeof data === 'string') {
```

Comment:
just typeof is sufficient here, no need for both checks. then you can make this a ternary